File: equals-transitive/Nan.java
class Nan {
    public static void main (String args[]) {
	double n = 0.0 / 0.0;
	System.out.println(java.lang.Math.min(n, 1.0));
	System.out.println(java.lang.Math.min(1.0, n));
    }
}

File: equals-transitive/nan.rb
a = 0.0 / 0.0
2.0.0-p247 :011 > [1.0, 2.0].min                                                                                                                                                     
 => 1.0 
2.0.0-p247 :012 > a
 => NaN 
2.0.0-p247 :013 > [1.0, a].min                                                                                                                                                       
ArgumentError: comparison of Float with 1.0 failed
